RuleDocumentation/UseIdenticalMandatoryParametersForDSC.md

```diff
@@ -4,13 +4,24 @@
 
 ## Description
 
-The `Get-TargetResource`, `Test-TargetResource` and `Set-TargetResource` functions of DSC Resource must have the same mandatory parameters.
+For script based DSC resources the `Get-TargetResource`, `Test-TargetResource` and `Set-TargetResource` functions must have identical mandatory parameters that are also the keys in the corresponding `mof` file.
```

Comment:
According to [this](https://msdn.microsoft.com/en-us/powershell/dsc/authoringresourcemof), the rule should be looking for all the properties in the mof file. 
>All three functions must take a parameter set that is identical to the set of properties defined in the MOF schema that you created for your resource. In this document, this set of properties is referred to as the “resource properties.”

It appears I had the wrong understanding about the rule :(. Guess I need to update the logic then, because the current logic only looks for keys that defined as identical mandatory parameters.